Source/FieldSolver/ElectrostaticSolver.cpp

```diff
@@ -363,28 +322,37 @@ WarpX::computePhi (const amrex::Vector<std::unique_ptr<amrex::MultiFab> >& rho,
             info.setMaxSemicoarseningLevel(max_semicoarsening_level);
             info.setSemicoarseningDirection(semicoarsening_direction);
         }
-        MLNodeTensorLaplacian linop( {Geom(lev)}, {boxArray(lev)}, {DistributionMap(lev)}, info );
-        linop.setBeta( beta_solver );
-#endif
-#else
-        // With embedded boundary: extract EB info
-        MLEBNodeFDLaplacian linop( {Geom(lev)}, {boxArray(lev)}, {DistributionMap(lev)}, info, {&WarpX::fieldEBFactory(lev)});
-
-#ifndef WARPX_DIM_RZ
-            // Note: this assumes that the beam is propagating along
-            // one of the axes of the grid, i.e. that only *one* of the Cartesian
-            // components of `beta` is non-negligible.
-            linop.setSigma({AMREX_D_DECL(
-                1._rt-beta[0]*beta[0], 1._rt-beta[1]*beta[1], 1._rt-beta[2]*beta[2])});
+
+#if defined(AMREX_USE_EB) || defined(WARPX_DIM_RZ)
+        // In the presence of EB or RZ: the solver assumes that the beam is
+        // propagating along  one of the axes of the grid, i.e. that only *one*
+        // of the components of `beta` is non-negligible.
+        MLEBNodeFDLaplacian linop( {Geom(lev)}, {boxArray(lev)}, {DistributionMap(lev)}, info
+#if defined(AMREX_USE_EB)
+            , {&WarpX::fieldEBFactory(lev)}
 #endif
+        );
+
+        // Note: this assumes that the beam is propagating along
+        // one of the axes of the grid, i.e. that only *one* of the
+        // components of `beta` is non-negligible.
+        linop.setSigma({AMREX_D_DECL(
+            1._rt-beta_solver[0]*beta_solver[0],
+            1._rt-beta_solver[1]*beta_solver[1],
+            1._rt-beta_solver[2]*beta_solver[2])});
 
         // if the EB potential only depends on time, the potential can be passed
         // as a float instead of a callable
         if (field_boundary_handler.phi_EB_only_t) {
             linop.setEBDirichlet(field_boundary_handler.potential_eb_t(gett_new(0)));
         }
         else linop.setEBDirichlet(field_boundary_handler.getPhiEB(gett_new(0)));
```

Comment:
Do we maybe need to wrap these lines into an `#if defined(AMREX_USE_EB) ... #endif` as well?
```suggestion
#if defined(AMREX_USE_EB)
        // if the EB potential only depends on time, the potential can be passed
        // as a float instead of a callable
        if (field_boundary_handler.phi_EB_only_t) {
            linop.setEBDirichlet(field_boundary_handler.potential_eb_t(gett_new(0)));
        }
        else linop.setEBDirichlet(field_boundary_handler.getPhiEB(gett_new(0)));
#endif
```